### DISCUSSION TITLE
Client-Side Console.log Messages Logged in Server Terminal When Directly Accessing Page

### DISCUSSION THREAD

**Discussion by ekiyak** [vercel/next.js#61296] [Help] [answered]
### Summary

Hello, I am using Next.js V14 with the "app router." In a server-side page under the "app" folder (for example, /user/profile), I am using a client-side component ('use client'). I'm logging a message with "console.log" in the client component to monitor whether it is working.

If I access the page through the navbar with "import Link from "next/link"," it works as expected, as described above. However, if I directly access the relevant page by entering the URL into the browser's address bar, the console.log message from the client-side component is logged in the console of the server terminal.

In this case, I am wondering, "Why was the console.log message from the client component logged in the server terminal?" I am asking this not as a problem but to better understand the process.

What could be the reason for this?

### Additional information

```js
--------------------PROFILE PAGE----------------------------------------------------------------------
import ClientComponent from "@/components/user/profile/clientcomponent"

const ProfilePage = ()=>{
    
    return(
        <>
            <ClientComponent/>
        </>
    )
}

export default ProfilePage
--------------------END PAGE----------------------------------------------------------------------


--------------------CLIENT COMPONENT----------------------------------------------------------
"use client"
const ClientComponent = ()=>{
    console.log('Client component is running ...')
    return(
        <>
            
        </>
    )
}

export default ClientComponent
-----------------END COMPONENT----------------------------------------------------------------
```

**Comment by Nadeera3784** [comment]
Hi @ekiyak,

When you enter the URL directly into the browser, Next.js server-side renders the page. This means the code for the page, including client-side components, runs on the server. That's why you see the console.log from your client component in the server terminal during server-side rendering (SSR).

However, when you navigate using <Link> from next/link, Next.js uses client-side navigation, avoiding a full page reload and SSR. As a result, the console.log in the client component runs in the browser, as you would expect.

This difference in behavior is just how Next.js works with rendering based on navigation types. It's important to know this to ensure your client-side code is compatible with server environments during SSR.

**Reply by ekiyak** [reply]
Thanks.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Unfortunately, `use client` is often confused with Client Side only. That's not the case.

`use client` is a hint to the the bundler, that the module has to be part of the JavaScript that' sent to the client. Server Components, never share their JavaScript with the client.

Client Components are then, the components we have used up till now, and they are rendered as part of SSR.

- https://github.com/reactwg/server-components/discussions/4

As explained above, when you navigate into the page, it does SPA transitions, by downloading (previously) the JS for that page, and rendering it client side.

When you arrive directly at the link, you trigger an SSR run, to serve the initial HTML to the browser, which can then use SPA transitions.

**Reply by ekiyak** [reply]
Thanks

**Comment by ekiyak** [comment]
Thank you for your responses. I understand better. Both of your replies are valid answers and I have to only choose one. Bye.
